src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1646,40 +1646,44 @@ bool Creature::CheckNoGrayAggroConfig(uint32 playerLevel, uint32 creatureLevel)
 
 float Creature::GetAttackDistance(Unit const* player) const
 {
+    // WoW Wiki: the minimum radius seems to be 5 yards, while the maximum range is 45 yards
+    float maxRadius = (45.0f * sWorld->getRate(RATE_CREATURE_AGGRO));
+    float minRadius = (5.0f * sWorld->getRate(RATE_CREATURE_AGGRO));
     float aggroRate = sWorld->getRate(RATE_CREATURE_AGGRO);
-    if (aggroRate == 0)
-        return 0.0f;
+    uint8 expansionMaxLevel = GetMaxLevelForExpansion(GetCreatureTemplate()->Expansion);
```

Comment:
```
src/server/game/Entities/Creature/Creature.cpp:1653:78:
 fatal error: no member named 'Expansion' in 'CreatureTemplate'
    uint8 expansionMaxLevel = GetMaxLevelForExpansion(GetCreatureTemplate()->Expansion);
                                                      ~~~~~~~~~~~~~~~~~~~~~  ^
1 error generated.
```

Reply:
wtf...

Reply:
Yes, I know. Sometimes the cause of an error is not as obvious as it would seem.

Reply:
Ye, I mean what happened to that field? where has it gone or how to I access it. Cant open VS right now to look for it

Reply:
There are 2 expansion fields `HealthScalingExpansion` (determines max health, our) and `RequiredExpansion` (client restricts interaction based on this, WDB field)

Reply:
which one would you recommend to use to identify the creature's expansion?

Reply:
`RequiredExpansion` should be the correct one in relation to when a creature was added to the game, but I could be wrong if this is about health vs level.